### PR TITLE
fix: Infinity scroll is slow

### LINKED ISSUE #5394: Infinity scroll is slow
### What happened?

Explore - Items - as I'm scrolling down, there are moments when the infinity scroll stops working and it seems like you got to the end of the list - even though you did not. 
Waiting helps, or scrolling to the top and back. But it should work much faster - this way someone new may think we have only a few NFTs for sale. 

https://user-images.githubusercontent.com/104380776/227983295-7bdba6cf-b7e5-4c50-a72f-06414be8258b.mov

### Please reproduce in steps

Explore - Items (bsx and ksm)

### Expected Behavior

to be fast

### What browsers are you seeing the problem on?

Desktop Chrome

### At which address did you encounter bug?

beta.kodadot.xyz, kodadot.xyz

### Are you logged in?

No

### Which wallet you are using?

PolkadotJS

### At which chain did you encounter bug?

_No response_

### Screenshots

-

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/transfer/?target=%3CMy_Kusama_Address_check_https://github.com/kodadot/nft-gallery/blob/main/CONTRIBUTING.md#creating-your-ksm-address%3E

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #5457: Collections view on RMRK2 showing only 20 cards
Seems is aware of 2453 items, can we load more?

image

### LINKED ISSUE #5461: Infinity scroll on explorer is broke
Hey I guess this might be resolved in ?
- #5429 
https://user-images.githubusercontent.com/5887929/229203045-4da031ba-b939-4f6d-86ef-da3afc7db5a8.mov

if not, let's fix it in separate PR, w/o this, explorer is non-usable.

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #5394
- [x] Closes #5461 
- [x] Closes #5457
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [ ] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸
tested on `/rmrk/explore/items?page=6`

### CODE REVIEW

**Review by roiLeo — COMMENTED** [review]
Would it not be better to integrate [nuxt-img](https://image.nuxtjs.org/components/nuxt-img) instead of `vue-lazyload`?

I ask this because I would like to avoid adding new plugins until we've done Nuxt3 migration

**Comment by Jarsen136** [comment]
> > Would it not be better to integrate [nuxt-img](https://image.nuxtjs.org/components/nuxt-img) instead of `vue-lazyload`?
> > I ask this because I would like to avoid adding new plugins until we've done Nuxt3 migration
> 
> Thanks. I will take a look.

I have tried to integrate nuxt-image into our project. It does works, but I found there still be something strange left to handler.

-  ipx provider https://image.nuxtjs.org/providers/ipx
-  full image url -> relative image url
- `nuxt.config.js`  the `target` config should be stay `static` instead of `server`
- related error:  
image


 I guess we could make another issue for someone else to pick it. Here is the demo PR
https://github.com/kodadot/nft-gallery/pull/5440

**Comment by roiLeo** [comment]
> I have tried to integrate nuxt-image into our project. It does works, but I found there still be something strange left to handler.

thank you for your investigation and sorry to see that it did not worked

> ipx provider https://image.nuxtjs.org/providers/ipx

Is this the solution for our case? Maybe custom provider is more in line with our needs?

> full image url -> relative image url

from what i understand images are served by the same domain `https://image(-beta).w.kodadot.xyz/...`

> `nuxt.config.js`  the `target` config should be stay `static` instead of `server`

yep this changes implies many modifications and could break stuff

> I guess we could make another issue for someone else to pick it. Here is the demo PR #5440

maybe #4426?


btw in #5440 I can see that you use nuxt-image in `libs/ui` or this folder shouldn't have access to nuxt (not installed over here)

**Review by roiLeo — COMMENTED** [review]
so you're using `v-lazy` in `libs/ui`? How would it worked if we remove this package from kodadot repo?

**Comment by vikiival** [comment]
Agree w @preschian 

@Jarsen136 please remove vue-lazy, we will make it in separated PR

**Comment by Jarsen136** [comment]
> > I have tried to integrate nuxt-image into our project. It does works, but I found there still be something strange left to handler.
> 
> thank you for your investigation and sorry to see that it did not worked
> 
> > ipx provider https://image.nuxtjs.org/providers/ipx
> 
> Is this the solution for our case? Maybe custom provider is more in line with our needs?
> 
Yes, it's the solution for the incorrect URL prefix.

> > full image url -> relative image url
> 
> from what i understand images are served by the same domain `https://image(-beta).w.kodadot.xyz/...`
> 
Yup, but when I set `static` to the target in `nuxt.config.js` file, the nuxt-image would not work with full image url like `https://image(-beta).w.kodadot.xyz/{id}`. It need the src to at the format like `/privoide/{id}`.

> > `nuxt.config.js`  the `target` config should be stay `static` instead of `server`
> 
> yep this changes implies many modifications and could break stuff
> 

> > I guess we could make another issue for someone else to pick it. Here is the demo PR #5440
> 
> maybe #4426?
> 
> btw in #5440 I can see that you use nuxt-image in `libs/ui` or this folder shouldn't have access to nuxt (not installed over here)

There are more details that could be continued by #4426

**Comment by Jarsen136** [comment]
> > Agree w @preschian
> > @Jarsen136 please remove vue-lazy, we will make it in separated PR
> 
> OK, I will make it.

Updated. I have removed the vue-lazyload plugin. Though scrolling process would be more smooth with the help of image lazyload.

**Comment by yangwao** [comment]
seems it's doing weird gltiches when scrolling down and returning me back? 

https://user-images.githubusercontent.com/5887929/229222786-fc4963d1-8a7f-4c12-b5c3-91d95fda4951.mov

**Comment by Jarsen136** [comment]
> seems it's doing weird gltiches when scrolling down and returning me back?

I could not reproduce that bug may because it would happen on a higher screen. And I make a fix based on that strange behaviour. Is it better now?

**Comment by Jarsen136** [comment]
Should fix all of these issues #5394 #5457 #5461.

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 1]
```diff
@@ -6,10 +6,12 @@
       'is-detail': isDetail,
     }">
     <img
-      class="is-block image-media__image"
       :src="src"
+      loading="lazy"
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply libs/ui/src/components/MediaItem/type/ImageMedia.vue]: ok

**Comment by vikiival** [comment]
Screenshot 2023-04-02 at 22 28 02


👀 You just used your one-per-month-force-push token

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 2]
```diff
@@ -20,6 +21,19 @@ defineProps<{
...
+const { $consola } = useNuxtApp()
+const { isDarkMode } = useTheme()
+const placeholder = computed(() => {
+  return isDarkMode.value ? '/placeholder.webp' : '/placeholder-white.webp'
+})
+
+const onError = (e: Event) => {
+  const target = e.target as HTMLImageElement
+  if (target) {
+    $consola.log('image load error', e)
```
Comment (vikiival) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:33]:
```suggestion
    $consola.log('[KODADOT::IMAGE] unable to load', e)
```

File: composables/useListInfiniteScroll.ts [hunk 3]
```diff
@@ -44,11 +44,11 @@ export default function ({
   useInfiniteScroll(
     containerRef,
     () => {
-      if (canLoadNextPage.value && total.value > 0) {
+      if (canLoadNextPage.value) {
         reachBottomHandler()
       }
     },
-    { distance: 1600 }
+    { distance: 0 }
```
Comment (vikiival) [inline composables/useListInfiniteScroll.ts:51]:
```
   * The minimum distance between the bottom of the element and the bottom of the viewport
   *
   * @default 0
   */
  distance?: number
  ```

File: composables/useListInfiniteScroll.ts [hunk 4]
```diff
@@ -35,7 +35,7 @@ export default function ({
   const scrollItemHeight = ref(300)
   const itemsPerRow = ref(4)
   const scrollItemSizeInit = ref(false)
-  const first = ref(defaultFirst || 20)
+  const first = ref(defaultFirst || 40)
```
Comment (vikiival) [inline composables/useListInfiniteScroll.ts:38]: Isn't this causing it too slow? 

Loading 40 items takes quuite a memory

Comment (Jarsen136) [reply composables/useListInfiniteScroll.ts:38]: I have discovered that the time required to fetch 20 items versus fetching 40 items is quite similar. Both of them take 500ms or so. Or I would limit it to 30. WDYT

File: composables/useListInfiniteScroll.ts (hunk 3 above)
Comment (Jarsen136) [reply composables/useListInfiniteScroll.ts]: Thanks, I have remove it.

**Comment by Jarsen136** [comment]
> Screenshot 2023-04-02 at 22 28 02
> 
> 👀 You just used your one-per-month-force-push token

My bad. It's because I pushed something unrelated commit to this branch by mistake. : )

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 5]
```diff
@@ -20,6 +21,19 @@ defineProps<{
   original: boolean
   isDetail?: boolean
 }>()
+const { $consola } = useNuxtApp()
+const { isDarkMode } = useTheme()
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:25]: This shouldn't be used here

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 6]
```diff
@@ -20,6 +21,19 @@ defineProps<{
...
+  return isDarkMode.value ? '/placeholder.webp' : '/placeholder-white.webp'
+})
+
+const onError = (e: Event) => {
+  const target = e.target as HTMLImageElement
+  if (target) {
+    $consola.log('[KODADOT::IMAGE] unable to load', e)
+    target.src = placeholder.value
+  }
+}
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:36]: I would move this at component level

**Review by roiLeo — CHANGES_REQUESTED** [review]

File: composables/useListInfiniteScroll.ts (hunk 4 above)
Comment (vikiival) [reply composables/useListInfiniteScroll.ts:38]: Ok if it's like that I am fine. 
Or let's do 30 and we will see

**Comment by exezbcz** [comment]
hmm, cannot open the deploy link, Am i doing something wrong? 
image

**Comment by vikiival** [comment]
@yangwao 👀

**Comment by yangwao** [comment]
> hmm, cannot open the deploy link, Am i doing something wrong?

https://deploy-preview-5429--koda-canary.netlify.app/ 

been changing deployments and old PRs might get wacky, `koda-nuxt` changed to `koda-canary`, at least know netlify retroactive doesn't update :p 

- #5420 

changing deployment while things are running 
image

**Comment by vikiival** [comment]
Lets fix @roiLeo stuff and let it roll

**Comment by yangwao** [comment]
wow amazing fix @Jarsen136 seems infinity scrolling is back to great state! :)
thank you!

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue (hunk 5 above)
Comment (Jarsen136) [reply libs/ui/src/components/MediaItem/type/ImageMedia.vue]: Yes, I have remove these stuff

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue (hunk 6 above)
Comment (Jarsen136) [reply libs/ui/src/components/MediaItem/type/ImageMedia.vue]: The error is throw directly from the img tag, so I guess placing an error handler on this component is not bad.

**Comment by Jarsen136** [comment]
> Lets fix @roiLeo stuff and let it roll

✅  Updated

**Review by vikiival — APPROVED** [review]

**Review by roiLeo — COMMENTED** [review]
why do you need to access `darkMode` in `libs/ui`?
same for `placeholder` I would pass it as props instead.

Use case: 
I'm using `@kodadot1/brick` in a different project and I want to use custom placeholder

**Comment by yangwao** [comment]
can we merge it 🥹

**Comment by vikiival** [comment]
Yup let it roll, and @roiLeo issue will be resolved in next one

**Pull request merged**